i3pystatus/pulseaudio/__init__.py

```diff
@@ -223,28 +220,33 @@ def sink_info_cb(self, context, sink_info_p, eol, _):
     def change_sink(self):
         sinks = list(s.split()[1] for s in self.sinks)
         if self.sink is None:
-            next_sink = sinks[(sinks.index(self.current_sink) + 1) %
-                              len(sinks)]
+            next_sink = sinks[(sinks.index(self.current_sink) + 1) % len(sinks)]
         else:
             next_sink = self.current_sink
 
         if self.move_sink_inputs:
-            sink_inputs = subprocess.check_output("pacmd list-sink-inputs".split(),
-                                                  universal_newlines=True)
+            sink_inputs = self._call_pactl(["list-sink-inputs"])
             for input_index in re.findall(r'index:\s+(\d+)', sink_inputs):
-                command = "pacmd move-sink-input {} {}".format(input_index, next_sink)
-
-                # Not all applications can be moved and pulseaudio, and when
-                # this fail pacmd print error messaging
-                with open(os.devnull, 'w') as devnull:
-                    subprocess.call(command.split(), stdout=devnull)
-        subprocess.call("pacmd set-default-sink {}".format(next_sink).split())
+                pactl_args = ["move-sink-input", input_index, next_sink]
+                self._call_pactl(pactl_args)
+        self._call_pactl(["set-default-sink", next_sink])
+
+    def _call_pactl(self, pactl_arguments):
+        try:
+            output = subprocess.check_output(
+                ["pactl"] + [str(arg) for arg in pactl_arguments],
+                universal_newlines=True)
+            return output
+        except Exception as exc:
+            self.logger.exception("Error while executing pactl")
+            self.output = {"color": self.color_error, "full_text": "Error while executing pactl"}
+            self.send_output()
```

Comment:
Every call of the functions `subprocess.check_output` or `subprocess.call` of this module were replaced in this PR by `self._call_pactl`.
This method call the `pactl` command with its arguments using `subprocess.check_output()` internally. The outputs are either treated by the callee or simply ignored.

If an error occurs, we update `self.output` with a minimal message and we notify the change using `self.send_output()`. The full stacktrace is available in the file logs (I checked) since `self.logger.exception` is now used.